app/modules/submissions/resources.py

```diff
@@ -244,3 +245,55 @@ def delete(self, submission):
         with context:
             db.session.delete(submission)
         return None
+
+
+@api.route('/tus/collect/<uuid:submission_guid>')
+@api.login_required(oauth_scopes=['submissions:read'])
+@api.response(
+    code=HTTPStatus.NOT_FOUND,
+    description='Submission not found.',
+)
+@api.resolve_object_by_model(Submission, 'submission', return_not_found=True)
+class SubmissionTusCollect(Resource):
+    """
+    Collect files uploaded by Tus endpoint for this Submission
+    """
+
+    @api.permission_required(
+        permissions.ObjectAccessPermission,
+        kwargs_on_request=lambda kwargs: {
+            'obj': kwargs['submission'],
+            'action': AccessOperation.WRITE,
+        },
+    )
+    @api.response(schemas.DetailedSubmissionSchema())
+    def get(self, submission):
+        submission, submission_guids = submission
+
+        if submission is None:
+            # We have checked the submission manager and cannot find this submission, raise 404 manually
+            raise werkzeug.exceptions.NotFound
+
+        repo, project = submission.ensure_repository()
+        updir = Submission.tus_upload_dir(submission.guid)
+        submission_abspath = submission.get_absolute_path()
+        submission_path = os.path.join(submission_abspath, '_submission')
+        ct = 0
+        for root, dirs, files in os.walk(updir):
+            ct = len(files)
```

Comment:
What is `ct` and is this meant to be a `+=`?

Reply:
nope just meant to be "are there any files (and if so how many)"